bundles/org.openhab.binding.juicenet/src/main/resources/OH-INF/thing/juicenet-device.xml

```diff
@@ -0,0 +1,271 @@
+		<label>Voltage</label>
+		<description>Current voltage. (V)</description>
+		<state readOnly="true" pattern="%d V"/>
+	</channel-type>
+
+	<channel-type id="wh_energy_type">
+		<item-type>Number:Energy</item-type>
+		<label>Current Energy</label>
+		<description>Current power level of vehicle. (Wh)</description>
+		<state readOnly="true" pattern="%.1f kWh"/>
+	</channel-type>
+
+	<channel-type id="savings_type">
+		<item-type>Number</item-type>
+		<label>Savings</label>
+		<description>Current session EV savings.</description>
+		<state readOnly="true" pattern="$%.2f"/>
+	</channel-type>
+
+	<channel-type id="watt_power_type">
+		<item-type>Number:Power</item-type>
+		<label>Power</label>
+		<description>Current charging power. (W)</description>
+		<state readOnly="true" pattern="%.2f kW"/>
+	</channel-type>
+
+	<channel-type id="seconds_charging_type">
+		<item-type>Number:Time</item-type>
+		<label>Charging Time</label>
+		<description>Charging time since plug-in time. (s)</description>
+		<state readOnly="true" pattern="%.0f min"/>
+	</channel-type>
+
+	<channel-type id="wh_energy_at_plugin_type">
+		<item-type>Number:Energy</item-type>
+		<label>Energy at Plugin</label>
+		<description>Energy value at the plugging time. (Wh)</description>
+		<state readOnly="true" pattern="%.2f kWh"/>
+	</channel-type>
+
+	<channel-type id="wh_energy_to_add_type">
+		<item-type>Number:Energy</item-type>
+		<label>Energy to Add</label>
+		<description>Amount of energy to be added in current session. (Wh)</description>
+		<state pattern="%.2f kWh"/>
+	</channel-type>
+
+	<channel-type id="lifetime_wh_energy_type">
+		<item-type>Number:Energy</item-type>
+		<label>Lifetime Energy</label>
+		<description>Total energy poured to vehicles during lifetime. (Wh)</description>
+		<state readOnly="true" pattern="%,.2f kWh"/>
+	</channel-type>
+
+	<channel-type id="lifetime_savings_type">
+		<item-type>Number</item-type>
+		<label>Lifetime Savings</label>
+		<description>EV driving saving during lifetime.</description>
+		<state readOnly="true" pattern="$%.2f"/>
+	</channel-type>
+
+	<channel-type id="gascost_type">
+		<item-type>Number</item-type>
+		<label>Gas Cost</label>
+		<description>Cost of gasoline used in savings calculations.</description>
+		<state readOnly="true" pattern="$%.2f"/>
+	</channel-type>
+
+	<channel-type id="mpg_type">
+		<item-type>Number</item-type>
+		<label>MPG</label>
+		<description>Miles per gallon used in savings calculations.</description>
+		<state readOnly="true" pattern="%.1f mpg"/>
+	</channel-type>
+
+	<channel-type id="ecost_type">
+		<item-type>Number</item-type>
+		<label>Utility Energy Cost</label>
+		<description>Cost of electricity from utility company. ($/kWh)</description>
+		<state readOnly="true" pattern="$%.2f"/>
+	</channel-type>
+
+	<channel-type id="whpermile_type">
+		<item-type>Number:Power</item-type>
+		<label>Watts per mile</label>
+		<description>Watts per mile.</description>
```

Comment:
Have you implemented UOM? See https://www.openhab.org/docs/concepts/units-of-measurement.html#defining-channeltypes

Not every country uses Miles so it should be possible to use UOM so it automatically changes depending on which country the openHAB is setup to be in.

Reply:
This one and others, such as miles per gallon, cannot be done without creating new UOM definitions on openhab_core.